features/mbedtls/targets/TARGET_CRYPTOCELL310/platform_alt.c

```diff
@@ -0,0 +1,57 @@
+ /*
+  *  platform_alt.c
+  *
+  *  Copyright (C) 2018, ARM Limited, All Rights Reserved
+  *  SPDX-License-Identifier: Apache-2.0
+  *
+  *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+  *  not use this file except in compliance with the License.
+  *  You may obtain a copy of the License at
+  *
+  *  http://www.apache.org/licenses/LICENSE-2.0
+  *
+  *  Unless required by applicable law or agreed to in writing, software
+  *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+  *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  *  See the License for the specific language governing permissions and
+  *  limitations under the License.
+  *
+  */
+
+#include "mbedtls/platform.h"
+#if defined(MBEDTLS_PLATFORM_SETUP_TEARDOWN_ALT)
+#include "sns_silib.h"
+
+/* once https://github.com/ARMmbed/mbedtls/issues/1200 will be supported,
+ * rndState should be part of mbedtls_platform_context
+ * Until then, we should keep it global and extern */
+
+CRYS_RND_State_t     rndState = { { 0 } } ;
+CRYS_RND_WorkBuff_t  rndWorkBuff = { { 0 } } ;
```

Comment:
Question: I think I understand what the problem is with the setters/getters, but why is it not possible to just put these two into the platform context and keep that global and extern?

Reply:
using the platform context is not according to design. 
`rndWorkBuff` is just a temporary sandbox, and wouldn't be in the platform context anyway
If we put the platform context as global, it is more likely it will be forgotten this way, rather than put `rndState` as gobal. If we put the pltform context as global, having it as parameter is redundant

Reply:
I dont follow. You quoted an issue that says we do not have a clean way of accessing the platform context from the hw acceleration code, but if we did then you would put rndState there. Do you mean that you do not want to make the platform context global? But if there was some sort of getter like suggested in the issue it would be "global" in a way. I dont really see why this is not in the platform context, but I am probably misunderstanding something

Reply:
Well, we haven't thought about the architecture yet.l Anyway, setter and getter doesn't make the context global, and keeps the abstraction of the context.